test/core/client_channel/lb_policy/xds_override_host_test.cc

```diff
@@ -213,6 +253,92 @@ TEST_F(XdsOverrideHostTest, ConnectingSubchannelIsQueued) {
   picker = ExpectState(GRPC_CHANNEL_READY);
   ExpectPickQueued(picker.get(), pick_arg);
 }
+
+TEST_F(XdsOverrideHostTest, DrainingState) {
```

Comment:
This test looks really good!

I think we also need a test similar to the `ConnectingSubchannelIsQueued` test above, but for DRAINING state.  We need to verify that we are still monitoring the subchannel's connectivity state even when the child has unreffed the subchannel and is not watching its connectivity state.

Reply:
Yes, I think that update is expected.  The way you've written the test, the address is starting out in a state other than DRAINING, and then you are switching it to state DRAINING.  This will trigger an update to the child policy that no longer contains this address, which will cause the child policy to return a new picker that will no longer return the removed address.